### ISSUE TITLE
Printing PDF mangles all the characters

### ISSUE THREAD

**Issue by amit777** [mozilla/pdf.js#10607]
Attach (recommended) or Link to PDF file here:
http://www.pdf995.com/samples/pdf.pdf

Configuration:
- Web browser and its version: Firefox 65.0.2 Quantum
- Operating system and its version: macOS Sierra 10.12.6
- PDF.js version: (Built in with firefox)
- Is a browser extension: No

Steps to reproduce the problem:
1.  Go to the URL 
2.  Print the document and choose "Save as PDF"

What is the expected behavior? (add screenshot)
![screen shot 2019-03-02 at 9 25 14 am](https://user-images.githubusercontent.com/2703309/53685259-22201180-3ccd-11e9-8e25-23f49ccde79e.png)

What went wrong? (add screenshot)
![screen shot 2019-03-02 at 9 26 07 am](https://user-images.githubusercontent.com/2703309/53685268-3cf28600-3ccd-11e9-98db-77187a451152.png)

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

**Comment by amit777** [comment]
Also, I want to clarify that PDFjs seems to render the PDF correctly in the browser.. Only when trying to Print->Save as PDF does it come out like this.

**Comment by Snuffleupagus** [comment]
Fixed upstream in https://bugzilla.mozilla.org/show_bug.cgi?id=1524640

**Comment by amit777** [comment]
Thank you!  I'm getting familiar with PDF.js and trying to figure out is there a known class of PDF rendering features that are not supported and don't plan to be supported?  I'm not sure if the information here is accurate, outdated or FUD:
https://www.pdftron.com/blog/pdf-js/pdf-js-incorrect-rendering/

**Comment by Snuffleupagus** [comment]
> I'm not sure if the information here is accurate, outdated or FUD:
> https://www.pdftron.com/blog/pdf-js/pdf-js-incorrect-rendering/

Based on a *quick* skim: Probably not very accurate, at least now. Most definitely very outdated, and I'm not sure if the "comparison" (I'm using that word loosely here...) provided has much value when the sole purpose of the article seem to be to get people to buy their software. ¯\_(ツ)_/¯

**Comment by THausherr** [comment]
Articles like that one are not a nice thing to do. It has no date (but based on their claiming one specific bug was not fixed, it is from 2015, as that one was fixed in Dec 2015), and PDF.js has the "disadvantage" that the bug tracker is accessible.

@amit777 annotations without appearance streams (most of those not generated by Adobe) is a weakness. I just tested the demo of pdftron, they are much better, although a few minor bugs.
